### DISCUSSION TITLE
Airflow 2.9.3: Unable to parse the connection string

### DISCUSSION THREAD

**Discussion by nitinmahawadiwar** [apache/airflow#43358] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### If "Other Airflow 2 version" selected, which one?

2.9.3

### What happened?

We are experiencing Azure Blob connection string parse error on Airflow 2.9.3 (self hosted, deployed through official helm chart). The same code is working fine on Auzre Managed Airflow 2.6.3. 
Both environments are using Python 3.8

`Broken DAG: [DAG Path]
Traceback (most recent call last):
File "/home/airflow/.local/lib/python3.8/site-packages/azure/storage/blob/_blob_service_client.py", line 195, in from_connection_string
account_url,secondary,credential=parse_connection_str(conn_str,credential,'blob')
File "/home/airflow/.local/lib/python3.8/site-packages/azure/storage/blob/_shared/base_client.py", line 387, in parse_connection_str
conn_str=conn_str.rstrip(",")
AttributeError: 'NoneType' object has no attribute 'rstrip'`

The code to get connection's details is as follow:
**We read a Variable. That variable stores the connection details (conn_id). This connection id exists in the Connections list.**

The Test connection was successfull from web ui.

**My query is**: Are there any changes from airflow 2.6.3 to airflow 2.9.3 over connection details (conn string, url format etc)

### What you think should happen instead?

The connection should work from the DAG as Test connection was successful.
 

### How to reproduce

Create a Azure Blog Storage connection type with all details, Test the connection from web UI.

Try to load connection from source code.

### Operating System

Debian GNU Linux 12

### Versions of Apache Airflow Providers

apache-airflow-providers-amazon          8.25.0
apache-airflow-providers-celery          3.7.2
apache-airflow-providers-cncf-kubernetes 8.3.3
apache-airflow-providers-common-io       1.3.2
apache-airflow-providers-common-sql      1.14.2
apache-airflow-providers-docker          3.12.2
apache-airflow-providers-elasticsearch   5.4.1
apache-airflow-providers-fab             1.2.2
apache-airflow-providers-ftp             3.10.0
apache-airflow-providers-google          10.21.0
apache-airflow-providers-grpc            3.5.2
apache-airflow-providers-hashicorp       3.7.1
apache-airflow-providers-http            4.12.0
apache-airflow-providers-imap            3.6.1
apache-airflow-providers-microsoft-azure 10.0.0
apache-airflow-providers-microsoft-winrm 3.4.0
apache-airflow-providers-mysql           5.6.2
apache-airflow-providers-odbc            4.6.2
apache-airflow-providers-openlineage     1.9.1
apache-airflow-providers-postgres        5.11.2
apache-airflow-providers-redis           3.7.1
apache-airflow-providers-sendgrid        3.5.1
apache-airflow-providers-sftp            4.10.2
apache-airflow-providers-slack           8.7.1
apache-airflow-providers-smtp            1.7.1
apache-airflow-providers-snowflake       4.1.0
apache-airflow-providers-sqlite          3.8.1
apache-airflow-providers-ssh             3.11.2

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

We are using AKS (Azure Kubernetes) 1.29 

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by potiuk** [comment]
You need to debug it. There were likely few 100s if not more changes between those versions. You can read release notes if you want to spend time on it, or someone who you ask here would have to do it, becasue there is no way to know it by heart.

And it's not clear where the error is - looks like you are getting somewhere None. Maybe because you have a typo or maybe something  is not accessing the right db or smth. You need to debug it.

Converted to a discussion - as this is not clear whether it's airflow issue.

**Reply by nitinmahawadiwar** [reply]
Thank you @potiuk , Let me debug & update you.
